### DISCUSSION TITLE
LC Tech 4Ch Relais - Tasmota - BMP 280 (alt. DHT22)

### DISCUSSION THREAD

**Discussion by RSV-Tuono** [arendst/Tasmota#11671] [Q&A] [answered]
Successfully flashed Tasmota on 4-CH Relay.
Perfect working with MQTT and Homeassistant.

Question: Is it possible to put an additional temperature sensor on the Relais-Board and send data over WiFi?

Thank you in advance for information.

**Comment by barbudor** [comment, marked as answer]
Yes. DS18x20 is supported by default on tasmota.bin and requires 1 GPIO.
More advanced I2C sensors such as BME280 or HDC1080 will require tasmota-sensors.bin (and 2 GPIO)

**Comment by RSV-Tuono** [comment]
Thank you @barbudor for trying to help!

Unfortunately there is a missunderstanding.
I have already running a BMP280 sensor and a 4Ch relay, too.

What I wanted to do is save an extra ESP8266, IP addresses running out of range :)

So I know that you need the tasmota-sensors.bin for BMP280,
and tasmota.bin for the LC TEch 4Ch relay, but:

I´m asking for a solution to put the BMP 280 in addition to the relay, with only one ESP8266.
There are some free pins, like RX TX GRD VCC on the relay board, maybe they are usable.
Also with a workaround for the binairies or GPIO settings in Config Menue.

Thank you very much for your help!

**Comment by sfromis** [comment]
No different binary needed, tasmota-sensors.bin-gz should work for both BMP280 and the 4CH relay board at the same time, and as long as you can find two free gpio pins, you can have a chain of I2C sensors connected, including support for two BME/BMP280 at the same time. In principle, TX/RX can be used (GND/VCC are not data pins), but some relay boards use TX/RX for talking to another MCU controlling the relays. Depends on which **exact** board you mean, which template you found in https://templates.blakadder.com/

If it is like https://templates.blakadder.com/LC-ESP01-4R-5V.html the ESP8266 sits on an ESP-01 module severely restricting available pins. Unless you want to solder to the tiny pins on the ESP8266 chip itself, the only option on that board would be to sacrifice the function of led and button, to use these pins for I2C.

**Reply by RSV-Tuono** [reply]
puhh, looks not as easy as it seems :)

@sfromis thank you very much for your detailed explanation!

Will give it a try and report if I got it working.

stay healthy!

Kind regards
RSV Tuono

**Comment by RSV-Tuono** [comment]
works like a charm!
![image](https://user-images.githubusercontent.com/28931337/114443161-a6902e80-9bcd-11eb-856a-80b1943a8255.png)

Is working with Original tasmota.bin

Only set RX GPIO3 to DS18x20
Red to 5V
Grey to GRD
Yellow to RX

thats it. 

Also thank you @barbudor . I read your post 3 times.. and now i know what you meant!

Great support!
